backend/src/modules/socket/gateways/socket.gateway.ts

```diff
@@ -0,0 +1,57 @@
+import { Logger, ValidationPipe } from '@nestjs/common';
+import { ConnectedSocket, MessageBody, OnGatewayConnection, OnGatewayDisconnect, OnGatewayInit, SubscribeMessage, WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
+import { SocketService } from '../socket.service';
+import { UserService } from '../../user/user.service';
+import { RedisService } from '../../redis/redis.service';
+import { RoomUserService } from '../../roomUser/roomUser.service';
+import { extractIdRoom } from '../../../common/utils/helper';
+
+@WebSocketGateway(3001, { cors: '*:*' })
```

Comment:
[EASY]
@MiraiJR Chổ này mà implement cái CORS vậy thì không an toàn nha em.
Này là em đang cho phép tất cả domain có thể connect tới server. Em nên giới hạn lại nha.
```typescript
@WebSocketGateway(3001, { cors: { origin: ['http://example1.com']}})
```